server/src/main/java/io/druid/server/coordinator/LoadQueuePeon.java

```diff
@@ -326,6 +310,21 @@ public void run()
     }
   }
 
+  public void start()
```

Comment:
Given that you have started this up.  You need to also stop it in the stop method.  You can do this by calling the same `scheduleAtFixedRate` method but with a `Callable` instead of a `Runnable`.  Then check some state and use that to return `Signal.REPEAT` versus `Signal.STOP`.